algorithmic_efficiency/workloads/mnist/workload.py

```diff
@@ -1,24 +1,70 @@
 """MNIST workload parent class."""
+
+import functools
+import itertools
 import math
 import os
-from typing import Dict
+from typing import Dict, Iterator, Optional
 
 from absl import flags
 from flax import jax_utils
 import jax
+import tensorflow as tf
+import tensorflow_datasets as tfds
 import torch
 import torch.distributed as dist
 
+from algorithmic_efficiency import data_utils
 from algorithmic_efficiency import spec
 import algorithmic_efficiency.random_utils as prng
 
 FLAGS = flags.FLAGS
 USE_PYTORCH_DDP = 'LOCAL_RANK' in os.environ
 
 
+def normalize(image, mean, stddev):
+  return (tf.cast(image, tf.float32) - mean) / stddev
+
+
+def get_mnist_dataset(data_rng: jax.random.PRNGKey,
+                      num_train_examples: int,
+                      train_mean,
+                      train_stddev,
+                      split: str,
+                      data_dir: str,
+                      global_batch_size: int):
+  shuffle = split in ['train', 'eval_train']
+  if shuffle:
+    tfds_split = f'train[:{num_train_examples}]'
+  elif split == 'validation':
+    tfds_split = f'train[{num_train_examples}:]'
+  else:
+    tfds_split = 'test'
+  ds = tfds.load(
+      'mnist:3.0.1', split=tfds_split, shuffle_files=False, data_dir=data_dir)
+  ds = ds.map(
+      lambda x: {
+          'inputs': normalize(x['image'], train_mean, train_stddev),
+          'targets': x['label'],
+      })
+  ds = ds.cache()
```

Comment:
given that `itertools.cycle` stores the entire ds in memory, we can delete this line